zeebe/restore/src/main/java/io/camunda/zeebe/restore/RestoreManager.java

```diff
@@ -92,20 +95,39 @@ private void logSuccessfulRestore(
   }
 
   private CompletableFuture<Void> restorePartition(
-      final RaftPartition partition, final long backupId, final boolean validateConfig) {
+      final InstrumentedRaftPartition partition,
+      final long backupId,
+      final boolean validateConfig) {
     final BackupValidator validator;
+    final RaftPartition raftPartition = partition.partition();
+
     if (validateConfig) {
       validator = new ValidatePartitionCount(configuration.getCluster().getPartitionsCount());
     } else {
       LOG.warn("Restoring without validating backup");
       validator = BackupValidator.none();
     }
-    return new PartitionRestoreService(backupStore, partition, new ChecksumProviderRocksDBImpl())
+
+    final var registry = partition.registry();
+    return new PartitionRestoreService(
+            backupStore, raftPartition, new ChecksumProviderRocksDBImpl())
         .restore(backupId, validator)
-        .thenAccept(backup -> logSuccessfulRestore(backup, partition.id().id(), backupId));
+        .thenAccept(backup -> logSuccessfulRestore(backup, raftPartition.id().id(), backupId))
+        .whenComplete(
+            (ok, error) -> {
+              registry.clear();
```

Comment:
:thought_balloon: this three calls are popping up all over the place :-D

Reply:
Go for it and add a helper in MicrometerUtil ;)